.github/workflows/stable-snapshot.yml

```diff
@@ -0,0 +1,97 @@
+    - cron: '0 16 * * 1' # every Monday at 9am PST
+  workflow_dispatch:
+    inputs:
+      dry_run:
+        type: boolean
+        default: false
+        description: Does not trigger job, but checks on whether the job should run.
+      force_snapshot:
+        description: 'Whether to force a snapshot. Useful if workflow is within 5 days of a ceremony'
+        required: false
+        default: false
+        type: boolean
+
+jobs:
+  check:
+    # This job checks whether snapshot should run.
+    runs-on: ubuntu-latest
+    outputs:
+      block_snapshot: ${{ steps.check.outputs.block_snapshot }}
+    env:
+      FORCE_SNAPSHOT: ${{ inputs.force_snapshot }}
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          fetch-depth: 0
+      - name: Determine whether to run a snapshot
+        id: check
+        shell: bash
+        run: |
+          set -euo pipefail
+
+          BRANCHES=$(git for-each-ref --format='%(refname:short)' | grep origin/ceremony/)
+          echo "${BRANCHES}"
+          # Check whether a ceremony was initiated within a week of the current date.
+          echo "block_snapshot=false" >> "${GITHUB_OUTPUT}"
+
+          # If we force a snapshot, exit early.
+          if [[ "${FORCE_SNAPSHOT}" ]]; then
+            exit
+          fi
+
+          ceremonyRegex="origin/ceremony/[0-9]{4}-[0-9]{2}-[0-9]{2}$"
+          for branch in ${BRANCHES}
+          do
+            if [[ "$branch" =~ ${ceremonyRegex} ]]; then
+              echo "found ceremony branch $branch"
+              branch_date=$(echo "${branch}" | cut -d '/' -f3)
+              days_diff=$(( ($(date -d "00:00" +%s) - $(date -d "${branch_date}" +%s)) / (24*3600) ))
+              if [[ "$days_diff" -lt 5 ]]; then
+                # Detected ceremony within 5 days of current date
+                echo "detected ceremony branch $branch within 5 days, stopping automated cron"
+                echo "block_snapshot=true" >> "${GITHUB_OUTPUT}"
+              fi
+            fi
+          done
+
+  run_snapshot_publish:
+    needs: check
+    if: (github.event_name == 'schedule' && github.repository == 'sigstore/root-signing' && needs.check.outputs.block_snapshot == 'false') || (github.event_name != 'schedule' && inputs.dry_run == false)  # Don't run workflow in forks on cron
+    permissions:
+      id-token: 'write'
+      issues: 'write'
+      pull-requests: 'write'
+      contents: 'write'
+    uses: sigstore/root-signing/.github/workflows/reuseable-snapshot-timestamp.yml@main
+    with:
+      snapshot_key: 'gcpkms://projects/sigstore-root-signing/locations/global/keyRings/root/cryptoKeys/snapshot'
+      repo: 'repository/'
+      branch: main
+      provider: 'projects/163070369698/locations/global/workloadIdentityPools/github-actions-pool/providers/github-actions-provider'
+      service_account: 'github-actions@sigstore-root-signing.iam.gserviceaccount.com'
+      timestamp: false
```

Comment:
yeah, again here - timestamp must be re-run if you renew a snapshot, so you might just want to omit this option and handle that timestamp must run inside the workflow defn if snapshot is true.

Reply:
Deleting this, there will be one extra workflow to generate timestamps midweek.